src/nupic/engine/Network.hpp

```diff
@@ -466,7 +462,14 @@ class Link;
   void resetEnabledPhases_();
 
   bool initialized_;
-  Collection<std::shared_ptr<Region>> regions_;
+	
+	/**
+	 * The list of regions registered with the Network.
+	 * Internally this is a map so it is easy to serialize
+	 * but externally this is a Collection object so it
+	 * retains API compatability.
```

Comment:
> The better API would use std::map instead, and internally Collection could have extended from std::map.

Collection contains both a vector of pairs and a map of indexes into the vector.  It would be difficult to extend from map and get a class that can also be indexed like a vector to get a pair. 

> ... for the sake of getting rid of Collection, ...

The Collection is still being used in several other places as well so it will take a while before we can get rid of it.  Lets let it live in the API for now.

Reply:
>  @breznak approved these changes

Thanks for doing the review for me.